staging/src/k8s.io/component-base/config/testing/helpers.go

```diff
@@ -0,0 +1,78 @@
+package testing
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+func RunTestsOnYAMLData(t *testing.T, scheme *runtime.Scheme, tests []TestCase, codecs serializer.CodecFactory) {
+	for _, rt := range tests {
+		t.Run(rt.name, func(t *testing.T) {
+			obj, err := decodeTestData(rt.in, scheme, rt.inGVK, codecs)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			const mediaType = runtime.ContentTypeYAML
+			info, ok := runtime.SerializerInfoForMediaType(codecs.SupportedMediaTypes(), mediaType)
+			if !ok {
+				t.Errorf("unable to locate encoder -- %q is not a supported media type", mediaType)
+			}
+
+			encoder := codecs.EncoderForVersion(info.Serializer, rt.outGV)
+
+			actual, err := runtime.Encode(encoder, obj)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			expected, err := ioutil.ReadFile(rt.out)
+			if err != nil && !os.IsNotExist(err) {
+				t.Fatalf("couldn't read test data: %v", err)
+			}
+
+			needsUpdate := false
+			if os.IsNotExist(err) {
+				needsUpdate = true
+				t.Error("couldn't find test data")
+			} else {
+				if !bytes.Equal(expected, actual) {
+					t.Errorf("the expected and actual output differs.\n\tin: %s\n\tout: %s\n\tgroupversion: %s\n\tdiff: \n%s\n",
+						rt.in, rt.out, rt.outGV.String(), cmp.Diff(string(expected), string(actual)))
+					needsUpdate = true
+				}
+			}
+			if needsUpdate {
+				const updateEnvVar = "UPDATE_COMPONENTCONFIG_TEST_DATA"
```

Comment:
@liggitt @thockin is there any naming convention for env vars like this?

Reply:
ones we use today:
```
const updateEnvVar = "UPDATE_EDIT_FIXTURE_DATA"
```
```
const updateEnvVar = "UPDATE_BOOTSTRAP_POLICY_FIXTURE_DATA"
```
```
const updateEnvVar = "UPDATE_COMPATIBILITY_FIXTURE_DATA"
```

Reply:
`UPDATE_<TEST-SPECIFIC-NAME>_FIXTURE_DATA`?

Reply:
Thanks for digging those up.
@tahsinrahman let's update to the convention @liggitt found

Reply:
updated to `UPDATE_COMPONENTCONFIG_FIXTURE_DATA`